plugins/terminal.koplugin/main.lua

```diff
@@ -1,57 +1,402 @@
+    }
+    self.shortcuts_menu = Menu:new{
+        show_parent = self.ui,
+        width = Screen:getWidth(),
+        height = Screen:getHeight(),
+        covers_fullscreen = true, -- hint for UIManager:_repaint()
+        is_borderless = true,
+        is_popout = false,
+        perpage = self.items_per_page,
+        onMenuHold = self.onMenuHoldShortcuts,
+        _manager = self,
+    }
+    table.insert(self.shortcuts_dialog, self.shortcuts_menu)
+    self.shortcuts_menu.close_callback = function()
+        UIManager:close(self.shortcuts_dialog)
+    end
+
+    -- sort the shortcuts:
+    if #self.shortcuts > 0 then
+        table.sort(self.shortcuts, function(v1, v2)
+            return v1.text < v2.text
+        end)
+    end
+    self:updateItemTable()
+end
+
+function Terminal:updateItemTable()
+    local item_table = {}
+    if #self.shortcuts > 0 then
+        local actions_count = 3 -- separator + actions
+        for nr, f in ipairs(self.shortcuts) do
+            local item = {
+                nr = nr,
+                text = f.text,
+                commands = f.commands,
+                editable = true,
+                deletable = true,
+                callback = function()
+                    -- so we know which middle button to display in the results:
+                    self.source = "shortcut"
+                    -- execute immediately, skip terminal dialog:
+                    self.command = self:ensureWhitelineAfterCommands(f.commands)
+                    Trapper:wrap(function()
+                        self:execute()
+                    end)
+                end
+            }
+            table.insert(item_table, item)
+            -- add page actions at end of each page with shortcuts:
+            local factor = self.items_per_page - actions_count
+            if nr % factor == 0 or nr == #self.shortcuts then
+                -- insert "separator":
+                table.insert(item_table, {
+                    text = " ",
+                    deletable = false,
+                    editable = false,
+                    callback = function()
+                        self:manageShortcuts()
+                    end,
+                })
+                -- actions:
+                self:insertPageActions(item_table)
+            end
+        end
+        -- no shortcuts defined yet:
+    else
+        self:insertPageActions(item_table)
+    end
+    local title = #self.shortcuts == 1 and _("Terminal shortcut") or _("Terminal shortcuts")
+    self.shortcuts_menu:switchItemTable(tostring(#self.shortcuts) .. " " .. title, item_table)
+    UIManager:show(self.shortcuts_dialog)
+end
+
+function Terminal:insertPageActions(item_table)
+    table.insert(item_table, {
+        text = "   " .. _("to terminal…"),
+        deletable = false,
+        editable = false,
+        callback = function()
+            self:terminal()
+        end,
+    })
+    table.insert(item_table, {
+        text = "   " .. _("close…"),
```

Comment:
```suggestion
        text = _("   close…"),
```

Reply:
That one should be okay with bidi and all that, plus it's actually nice not to bother the translators with how many spaces to add. :-)